pandas/core/dtypes/common.py

```diff
@@ -510,6 +510,39 @@ def is_interval_dtype(arr_or_dtype):
     return IntervalDtype.is_dtype(arr_or_dtype)
 
 
+def is_interval_arraylike(arr):
```

Comment:
This may not be needed in the long-term; once #19453 is closed `is_interval_dtype` should be sufficient anywhere I've used this, so haven't added related tests.  Can add tests if this PR looks like it will get closed first.  Or is there a reason we'd want to keep this after #19453, in which case tests should be added regardless?

Reply:
sure as part of adding first class dtypes we would want to remove this, but no problem add it here now. you can add a couple of basic test for this though.